src/Engine/Spells/CastSpellInfo.cpp

```diff
@@ -222,96 +239,114 @@ void CastSpellInfoHelpers::CastSpell() {
+            pSpellSprite.spell_target_pid = spell_targeted_at;
+            pSpellSprite.field_60_distance_related_prolly_lod = target_direction.uDistance;
+            pSpellSprite.uFacing = target_direction.uYawAngle;
+            memcpy(&pSpellSprite.containing_item, &pPlayer->pInventoryItemList[pPlayer->pEquipment.uMainHand - 1], sizeof(pSpellSprite.containing_item));
+            // &pParty->pPlayers[pCastSpell->uPlayerID].spellbook.pDarkSpellbook.bIsSpellAvailable[36
+            // *
+            // pParty->pPlayers[pCastSpell->uPlayerID].pEquipment.uMainHand +
+            // 5], );
+            pSpellSprite.uSectorID = pIndoor->GetSector(pSpellSprite.vPosition);
+            pSpellSprite.uAttributes |= SPRITE_MISSILE;
+            if (pParty->bTurnBasedModeOn) {
+                pSpellSprite.uAttributes |= SPRITE_HALT_TURN_BASED;
+            }
+            if (pSpellSprite.Create(target_direction.uYawAngle, target_direction.uPitchAngle, pObjectList->pObjects[pSpellSprite.uObjectDescID].uSpeed,
+                        pCastSpell->uPlayerID + 1) != -1 && pParty->bTurnBasedModeOn) {
+                ++pTurnEngine->pending_actions;
+            }
+        } else if (pCastSpell->uSpellID == SPELL_WATER_TOWN_PORTAL) {
+            if (pPlayer->CanCastSpell(uRequiredMana)) {
+                int amount = 10 * spell_level;
+                if (pParty->uFlags & (PARTY_FLAGS_1_ALERT_RED |
+                            PARTY_FLAGS_1_ALERT_YELLOW) &&
+                        spell_mastery != PLAYER_SKILL_MASTERY_GRANDMASTER ||
+                        grng->Random(100) >= amount && spell_mastery != PLAYER_SKILL_MASTERY_GRANDMASTER) {
+                    SpellFailed(pCastSpell, LSTR_SPELL_FAILED);
+                    continue;
+                }
+                TownPortalCasterId = pCastSpell->uPlayerID;
+                pMessageQueue_50CBD0->AddGUIMessage(UIMSG_OnCastTownPortal, 0, 0);
+                spell_sound_flag = true;
+            }
+        } else if (pCastSpell->uSpellID == SPELL_WATER_LLOYDS_BEACON) {
+            if (pCurrentMapName == "d05.blv") {  // Arena
+                SpellFailed(pCastSpell, LSTR_SPELL_FAILED);
+                continue;
+            }
+            if (pPlayer->CanCastSpell(uRequiredMana)) {
+                pEventTimer->Pause();
+                pMessageQueue_50CBD0->AddGUIMessage(UIMSG_OnCastLloydsBeacon, 0, 0);
+                _506348_current_lloyd_playerid = pCastSpell->uPlayerID;
+                ::uRequiredMana = uRequiredMana;
+                ::sRecoveryTime = sRecoveryTime;
+                LloydsBeaconSpellLevel = (int)(604800 * spell_level);
+                LloydsBeaconSpellId = pCastSpell->uSpellID;
+                pCastSpell->uFlags |= ON_CAST_NoRecoverySpell;
+                pMessageQueue_50CBD0->AddGUIMessage(UIMSG_OnCastLloydsBeacon, 0, 0);
+            }
+        } else {
+        int amount;
+        Vec3i spell_velocity;
+
+        spell_velocity.x = 0;
```

Comment:
I believe it's zero-initialized by default. If you want to spell it out explicitly, you can just declare it as `Vec3i spell_velocity = Vec3i(0, 0, 0);`